sql/sys_vars.cc

```diff
@@ -3826,6 +3826,15 @@ static bool fix_rpl_semi_sync_master_wait_point(sys_var *self, THD *thd,
   return false;
 }
 
+static bool fix_rpl_semi_sync_master_wait_for_slave_count
+  (sys_var *self, THD *thd, enum_var_type type)
+{
+  mysql_mutex_unlock(&LOCK_global_system_variables);
```

Comment:
I wonder how important this is, exactly?

Reply:
It's still a good idea, as `repl_semisync_master.clear_acked_tranx_nodes();` takes a lock of its own. Looking through other functions, it looks like the _only_ function which doesn't release `LOCK_global_system_variables` is `update_slave_skip_counter` - I wonder if that is intentional, or if it is vulnerable to deadlock/slowing down other connections which are updating variables.